Style: Optimize Transfer feature UI and components

yikZero opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Style**
  * Improved layout and spacing of transfer home screen for better readability.
  * Enhanced QR code section with clearer theming, borders, and descriptive text.
  * Updated pairing code display, making it more prominent and visually consistent.
  * Refined server status bar with a new custom layout and simplified interaction for copying server URLs.
  * Improved form labeling and responsiveness for entering pairing codes, including dynamic button sizing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yikZero committed:
style: enhance PrimeTransferServerStatusBar component for improved status display

- Replaced Alert component with a more flexible XStack layout for server status display.
- Updated status retrieval logic to use getStatusInfo function, consolidating state handling.
- Improved UI elements with dynamic styling based on connection state, including custom server handling.
- Streamlined code by removing unused props and simplifying the rendering logic.

yikZero committed:
refactor: enhance PrimeTransferHomeQrCode component with improved layout and theming

- Updated the layout of the PrimeTransferHomeQrCode component to utilize a more structured YStack for better organization.
- Introduced dynamic theming with useThemeVariant for improved visual consistency based on light/dark mode.
- Enhanced QR code display with a new styling approach, including a background stack and updated skeleton loading states.
- Simplified button rendering and improved accessibility by ensuring proper alignment and spacing.
- Removed unused imports and streamlined the component for better readability and maintainability.

yikZero committed:
refactor: update PrimeTransferHome and PrimeTransferHomeEnterLink components for improved layout and localization

- Enhanced the layout of PrimeTransferHome by replacing Stack with YStack for better spacing and organization.
- Adjusted gap size in Stack for improved visual consistency.
- Updated PrimeTransferHomeEnterLink to utilize Form.Field for label and description, improving accessibility and clarity.
- Added new translation key 'swap_revoke_successful' to support localization across multiple languages.
- Updated localization files for various languages to include the new translation key.

yikZero committed:
Merge branch 'x' into prime_custom_server_ui

revan-zhang commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/ba6b3486-03a5-479b-839a-994a7eb0431f?refs=85fae12476ba718fac4fa9c0457e040c34dc9f13&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/ba6b3486-03a5-479b-839a-994a7eb0431f/license?refs=85fae12476ba718fac4fa9c0457e040c34dc9f13&source=prChecksComment)

sidmorizon commented on packages/kit/src/views/Prime/pages/PagePrimeTransfer/components/PrimeTransferServerStatusBar.tsx:
```diff
@@ -145,24 +121,51 @@ export function PrimeTransferServerStatusBar() {
+        size="$bodyMd"
+        color="$text"
+        {...(statusInfo?.isCustomServer && {
+          onPress: handleTextPress,
```
之前这行字后面有个复制 icon，现在貌似没有，用户不一定知道这里可以点击复制 URL

yikZero replied:
这属于额外附加功能，不想这块太显眼。有需要的会尝试点击。不知道的进管理也能复制

sidmorizon commented on packages/kit/src/views/Prime/pages/PagePrimeTransfer/components/PrimeTransferHomeQrCode.tsx:
```diff
@@ -109,18 +141,9 @@ export function PrimeTransferHomeQrCode() {
-                </>
-              )}
+            <YStack>
+              <Skeleton.BodyMd w={230} />
```
大屏下也是双行骨架屏的话，会导致加载完成后屏幕文字跳跃（配对码只有一行）

yikZero replied:
QrCode 区域这个我限制了最大宽度 400px，会保证两行显示。

sidmorizon commented on packages/kit/src/views/Prime/pages/PagePrimeTransfer/components/PrimeTransferServerStatusBar.tsx:
```diff
@@ -145,24 +121,51 @@ export function PrimeTransferServerStatusBar() {
+      </SizableText>
+
+      <IconButton
+        icon="ServerOutline"
```
之前没用 IconButton 的原因是因为这个按钮看不出来可以点击，用户可能单纯以为只是个静态图标

yikZero replied:
那这块我给调整成文字按钮

yikZero committed:
refactor: replace IconButton with Button in PrimeTransferServerStatusBar for improved accessibility

- Updated the PrimeTransferServerStatusBar component to replace the IconButton with a Button for better usability.
- Enhanced the button to include a localized label using the intl.formatMessage function, improving user experience and accessibility.

yikZero committed:
Merge branch 'x' into prime_custom_server_ui

yikZero committed:
refactor: enhance PrimeTransferHomeQrCode and PrimeTransferServerStatusBar components for improved layout and accessibility

- Updated PrimeTransferHomeQrCode to utilize useMedia for responsive design and improved skeleton loading states.
- Simplified button rendering logic to conditionally display a loading skeleton or the button based on state.
- Refactored PrimeTransferServerStatusBar layout for better alignment and spacing, ensuring consistent styling.
- Enhanced accessibility by ensuring proper props are passed to components and improving button usability.

yikZero committed:
Merge branch 'x' into prime_custom_server_ui

sidmorizon committed:
Merge branch 'x' into prime_custom_server_ui

sidmorizon committed:
Merge branch 'x' into prime_custom_server_ui
